padertorch/data/segment.py

```diff
@@ -0,0 +1,413 @@
+
+        if any([not isinstance(value, np.ndarray) for value in to_segment]):
+            raise ValueError(
+                'This segmenter only works on numpy arrays',
+                'However, the following keys point to other types:',
+                '\n'.join([f'{key} points to a {type(to_segment[idx])}'
+                           for idx, key in enumerate(to_segment_keys)])
+            )
+
+        to_segment_lengths = [
+            v.shape[axis[i]] for i, v in enumerate(to_segment)]
+        assert to_segment_lengths[1:] == to_segment_lengths[:-1], (
+            'The shapes along the segment dimension of all entries to segment'
+            ' must be equal!\n'
+            f'segment keys: {to_segment_keys}'
+            f'to_segment_lengths: {to_segment_lengths}'
+        )
+        assert len(to_segment) > 0, ('Did not find any signals to segment',
+                                     self.include, self.exclude, to_segment)
+        to_segment_length = to_segment_lengths[0]
+
+        # Discard examples that are shorter than `length`
+        if to_segment_length < self.length:
+            import lazy_dataset
+            raise lazy_dataset.FilterException()
+
+        boundaries, segmented = self.segment(to_segment, to_segment_length,
+                                             axis=axis)
+
+        segmented_examples = list()
+        if self.copy_keys is not None:
+            example = {key: example.pop(key) for key in self.copy_keys}
+        for idx, (start, stop) in enumerate(boundaries):
+            example_copy = copy(example)
+            example_copy.update({to_segment_keys[i]: value[idx]
+                                 for i, value in enumerate(segmented)})
+
+            example_copy.update(segment_start=start, segment_stop=stop)
+            segmented_examples.append(deflatten(example_copy))
+        return segmented_examples
+
+
+    def segment(self, to_segment, to_segment_length, axis):
+
+        anchor = get_anchor(
+            to_segment_length, self.length, self.shift,
+            mode=self.anchor_mode, rng=self.rng
+        )
+
+        boundaries = get_segment_boundaries(
+            to_segment_length, self.length, self.shift,
+            anchor=self.anchor_mode, rng=self.rng
+        )
+
+        segmented = [segment(signal, length=self.length, shift=self.shift,
+                       rng=self.rng, axis=axis[i], anchor=anchor)
+                     for i, signal in enumerate(to_segment)]
+        return boundaries, segmented
+
+    def get_to_segment_keys(self, example):
+        if self.include is None:
+            return [
+                key for key, value in example.items()
+                if not key in self.exclude and isinstance(value, np.ndarray)
+            ]
+        else:
+            return [
+                key for key in example.keys()
+                if not key in self.exclude and
+                any([include_key in key for include_key in self.include])
+            ]
```

Comment:
What if a key is in `self.include` but not in the example? Should the code pass or should it raise an exception?

Reply:
It should probably raise an exception, I can add that.